lib/rust/profiler/flame-graph/src/lib.rs

```diff
@@ -6,38 +6,40 @@
 #![deny(non_ascii_idents)]
 #![warn(unsafe_code)]
 
-use crate::State::Active;
-use crate::State::Paused;
-
 use enso_profiler as profiler;
 use enso_profiler_data as data;
 
 
+
+// =================
+// === Constants ===
+// =================
+
+pub const COLOR_BLOCK_ACTIVE: &str = "flame_graph_block_color_active";
+pub const COLOR_BLOCK_PAUSED: &str = "flame_graph_block_color_paused";
+pub const COLOR_MARK_DEFAULT: &str = "flame_graph_mark_color";
+
+
+type RowNumber = i32;
+
+
 // ==================
 // === Block Data ===
 // ==================
 
-/// Indicates whether a block indicates an active interval or a paused interval. Paused intervals
-/// are used to represent async tasks that are started and awaited, but that have made no progress.
-#[derive(Copy, Clone, Debug)]
-pub enum State {
-    Active,
-    Paused,
-}
-
 /// A `Block` contains the data required to render a single block of a frame graph.
 #[derive(Clone, Debug)]
 pub struct Block {
     /// Start x coordinate of the block.
-    pub start: f64,
+    pub start:       f64,
     /// End x coordinate of the block.
-    pub end:   f64,
+    pub end:         f64,
     /// Row that the block should be placed in.
-    pub row:   u32,
+    pub row:         RowNumber,
     /// The label to be displayed with the block.
-    pub label: String,
-    /// Indicates what state this block represents (active/paused).
-    pub state: State,
+    pub label:       String,
+    /// Indicates the name of the color for lookup in the theme.
+    pub theme_color: &'static str,
```

Comment:
Excellent explanation @kazcw! Thanks for writing it.

Reply:
This is an excellent approach! I made the changes. There are a few tweaks that deviate from your proposal, for example, not using `From` due to orphan rule issues, but a new trait, which seemed nicer than a newtype to make the conversion work. But have a look and let me know if you think that can still be improved. 

> This would require fixing a bit of a hack from my original implementation of hybrid graph: There shouldn't be a hybrid graph type containing blocks arranged according to different disciplines; it would be better for the render-profile scene to draw a callgraph, draw a rungraph under it, and draw the new 1-bar-high frametime graph in the stack too. I just did it the current way because it was easier due to my inexperience with Enso's FRP networks.

Adding `Sections` like this is on my agenda for the visualization, but I think here it is not yet needed. This change will probably come for the multiprocess visualization.